diesel/src/pg/expression/expression_methods.rs

```diff
@@ -683,3 +683,592 @@ where
+     * #
+     * # table! {
+     * #     hosts {
+     * #         id -> Integer,
+     * #         address -> Inet,
+     * #     }
+     * # }
+     * #
+     * # fn main() {
+     * #     run_test().unwrap();
+     * # }
+     * #
+     * # fn run_test() -> QueryResult<()> {
+     * #     use self::hosts::dsl::*;
+     * #     use ipnetwork::IpNetwork;
+     * #     use std::str::FromStr;
+     * #     let conn = establish_connection();
+     * #     conn.execute("DROP TABLE IF EXISTS hosts").unwrap();
+     * #     conn.execute("CREATE TABLE hosts (id SERIAL PRIMARY KEY, address INET NOT NULL)").unwrap();
+     * diesel::insert_into(hosts)
+     *     .values(vec![address.eq(IpNetwork::from_str("10.0.2.3").unwrap())])
+     *     .execute(&conn)?;
+     *
+     * let addr = hosts
+     *     .select(address.and(IpNetwork::from_str("0.0.0.255").unwrap()))
+     *     .first::<IpNetwork>(&conn)?;
+     * assert_eq!(addr, IpNetwork::from_str("0.0.0.3").unwrap());
+     * #     Ok(())
+     * # }
+     * ```
+     */
+    fn and<T>(self, other: T) -> dsl::AndNet<Self, T>
+    where
+        T: AsExpression<Inet>,
+    {
+        Grouped(AndNet::new(self, other.as_expression()))
+    }
+
+    /**
+     * Creates a PostgreSQL `|` expression.
+     *
+     * This operator computes the bitwise OR between two network addresses.
+     *
+     * # Example
+     *
+     * ```rust
+     * # #[macro_use] extern crate diesel;
+     * # extern crate ipnetwork;
+     * # include!("../../doctest_setup.rs");
+     * #
+     * # table! {
+     * #     hosts {
+     * #         id -> Integer,
+     * #         address -> Inet,
+     * #     }
+     * # }
+     * #
+     * # fn main() {
+     * #     run_test().unwrap();
+     * # }
+     * #
+     * # fn run_test() -> QueryResult<()> {
+     * #     use self::hosts::dsl::*;
+     * #     use ipnetwork::IpNetwork;
+     * #     use std::str::FromStr;
+     * #     let conn = establish_connection();
+     * #     conn.execute("DROP TABLE IF EXISTS hosts").unwrap();
+     * #     conn.execute("CREATE TABLE hosts (id SERIAL PRIMARY KEY, address INET NOT NULL)").unwrap();
+     * diesel::insert_into(hosts)
+     *     .values(vec![address.eq(IpNetwork::from_str("10.0.2.3").unwrap())])
+     *     .execute(&conn)?;
+     *
+     * let addr = hosts
+     *     .select(address.or(IpNetwork::from_str("0.0.0.255").unwrap()))
+     *     .first::<IpNetwork>(&conn)?;
+     * assert_eq!(addr, IpNetwork::from_str("10.0.2.255").unwrap());
+     * #     Ok(())
+     * # }
+     * ```
+     */
+    fn or<T>(self, other: T) -> dsl::OrNet<Self, T>
+    where
+        T: AsExpression<Inet>,
+    {
+        Grouped(OrNet::new(self, other.as_expression()))
+    }
+}
+
+impl<T> PgNetExpressionMethods for T
+where
+    T: Expression,
+    T::SqlType: InetOrCidr,
+{
+}
+
+/// PostgreSQL specific methods present between CIDR/INET expressions and Bigint expressions
+pub trait PgNetAddExpressionMethods: Expression + Sized {
```

Comment:
Should this not just implement the corresponding add operator? So that `column_a + column_b` just works there.

Reply:
Yup, definitely, I actually was not aware that Diesel supported this.

Reply:
Done in d486715c2